src/Sarif.Viewer.VisualStudio/Tags/SarifLocationTagger.cs

```diff
@@ -214,21 +243,16 @@ public static void RemoveAllTags()
         /// </summary>
         public static void DisposeStaticObjects()
         {
-            List<SarifLocationTag> tagsToDispose = new List<SarifLocationTag>();
-
-            using (TagListLock.EnterWriteLock())
+            List<SarifLocationTagger> taggersToDispose;
+            using (SarifTaggersLock.EnterWriteLock())
             {
-                // There is no need to collect the tags from the run Id to SARIF tag dictionary as the
-                // objects exist in both dictionaries and only need to be disposed once.
-                tagsToDispose.AddRange(SourceCodeFileToSarifTags.Values.SelectMany(sarifTags => sarifTags));
-
-                SourceCodeFileToSarifTags.Clear();
-                RunIdToSarifTags.Clear();
+                taggersToDispose = new List<SarifLocationTagger>(SarifTaggers);
```

Comment:
Why duplicate instead of swap out?

Reply:
The number of taggers is going to be "about" equal to the number of documents the user has opened. So that number is small, so I'd prefer to leave this as a static readonly list.

---
In reply to: [497151946](https://github.com/microsoft/sarif-visualstudio-extension/pull/206#discussion_r497151946) [](ancestors = 497151946)